### PR TITLE
ruby: do not depend on host make, setup $RUBYLIB

### LINKED ISSUE #239: Ruby bundler can't build native extensions
Using any devenv version after nakedshell is merged causes ruby to not build any native extensions.  They all fail while 'creating Makefile'.  I checked the example project for naked shell here: https://github.com/blrB/hanami-realworld-example-app/pull/1/files This also fails to bundle install for me.  
Here's how it fails:
```
Installing bcrypt 3.1.18 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
/nix/store/ckf8k2sqhwgxm0xbncbif2jh5f67d2zi-ruby-2.7.6/bin/ruby -I /nix/store/ckf8k2sqhwgxm0xbncbif2jh5f67d2zi-ruby-2.7.6/lib/ruby/2.7.0 extconf.rb
creating Makefile

current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\= sitearchdir\=./.gem.20221223-3398809-dq26y6 sitelibdir\=./.gem.20221223-3398809-dq26y6 clean
current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\= sitearchdir\=./.gem.20221223-3398809-dq26y6 sitelibdir\=./.gem.20221223-3398809-dq26y6
make failedNo such file or directory - make

Gem files will remain installed in /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18 for inspection.
Results logged to /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/extensions/x86_64-linux/2.7.0/bcrypt-3.1.18/gem_make.out

An error occurred while installing bcrypt (3.1.18), and Bundler cannot continue.
Make sure that `gem install bcrypt -v '3.1.18' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  bcrypt
```
Here's the contents of gem_make.out:
```
current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
/nix/store/ckf8k2sqhwgxm0xbncbif2jh5f67d2zi-ruby-2.7.6/bin/ruby -I /nix/store/ckf8k2sqhwgxm0xbncbif2jh5f67d2zi-ruby-2.7.6/lib/ruby/2.7.0 extconf.rb
creating Makefile

current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\= sitearchdir\=./.gem.20221223-3398809-dq26y6 sitelibdir\=./.gem.20221223-3398809-dq26y6 clean
current directory: /home/cole/Projects/github.com/hanami-realworld-example-app/.devenv/state/.bundle/ruby/2.7.0/gems/bcrypt-3.1.18/ext/mri
make DESTDIR\= sitearchdir\=./.gem.20221223-3398809-dq26y6 sitelibdir\=./.gem.20221223-3398809-dq26y6
make failedNo such file or directory - make
```

This is the project I've been testing with:
https://gist.github.com/thecatwasnot/77fdc70bf08686f29a8a9d6f8dbd48a6

**Version**
 
$ devenv version
devenv: 0.5

My NixOS repo: https://github.com/thecatwasnot/nixfiles

I have noticed that I'm not seeing some environment variables inside the devenv shell that might be relevant, for example, no RUBYLIB.

### PR DESCRIPTION
**Pull request by thenonameguy**
fixes #239

### CODE REVIEW

**Comment by thenonameguy** [comment]
[@thecatwasnot](https://github.com/thecatwasnot) please verify that this solves your use-cases.

File: src/modules/languages/ruby.nix [hunk 1]
```diff
@@ -19,15 +19,20 @@ in
...
 
     env.BUNDLE_PATH = config.env.DEVENV_STATE + "/.bundle";
 
     env.GEM_HOME = "${config.env.BUNDLE_PATH}/${cfg.package.rubyEngine}/${cfg.package.version.libDir}";
 
-    enterShell = ''
-      export GEM_PATH="$GEM_HOME/gems:$GEM_PATH"
-      export PATH="$GEM_HOME/bin:$PATH"
-    '';
+    enterShell =
```
Comment (domenkozar) [inline src/modules/languages/ruby.nix:29]: Let's set these via `env` instead, so they show up in `devenv info`

**Review by domenkozar — CHANGES_REQUESTED** [review]

File: src/modules/languages/ruby.nix (hunk 1 above)
Comment (thenonameguy) [reply src/modules/languages/ruby.nix:29]: AFAIK currently you cannot do env var prepends with the current system, just setting. devshell has this syntactically in it's env var implementation.

**Comment by thecatwasnot** [comment]
Thanks guys for a quick turnaround!  This change and adding gcc fixed the issue.  As far as providing the compiler, could we either add it in the module or in the example?  
The errors that rubygems spits out arn't super helpful if gcc is completely absent, and therefore it's tricky to work out that's the problem. 
On the other side I'm not sure what the risks are of providing it in the module.  I think providing the example as documentation is probably sufficient.

**Comment by thenonameguy** [comment]
> Thanks guys for a quick turnaround!  This change and adding gcc fixed the issue.  As far as providing the compiler, could we either add it in the module or in the example?  
> 
> The errors that rubygems spits out arn't super helpful if gcc is completely absent, and therefore it's tricky to work out that's the problem. 
> 
> On the other side I'm not sure what the risks are of providing it in the module.  I think providing the example as documentation is probably sufficient.

It's a tradeoff between full reproducibility (and large devenv profile size ~= closure size in nix terms) and language module specificity (only having $LANG specific tooling added/configured). 
We moved a bit in the direction to the former in this case.
Probably we could just add 'pkgs.stdenv' to 'packages' on a language by language basis, whether it needs the normal C compilation toolchain when invoked within a 'devenv shell'.

**Comment by domenkozar** [comment]
That's probably a good idea: https://discourse.nixos.org/t/rubypackages-rails-has-been-removed-from-nix-package-store-starting-22-05/24184/8

**Pull request merged**